File: project/block_manager/services/nodes/frameworks/pytorch/nodes/softmax.py
"""PyTorch Softmax Activation Node Definition"""

from typing import Any, Dict, List, Optional

from ....core.types import ConfigField, Framework, NodeMetadata, TensorShape
from ..base import PyTorchPassthroughNode


class SoftmaxNode(PyTorchPassthroughNode):
    """Softmax activation function"""

    @property
    def metadata(self) -> NodeMetadata:
        return NodeMetadata(
            type="softmax",
            label="Softmax",
            category="basic",
            color="var(--color-accent)",
            icon="Function",
            description="Softmax activation function",
            framework=Framework.PYTORCH
        )

    @property
    def config_schema(self) -> List[ConfigField]:
        return [
            ConfigField(
                name="dim",
                label="Dimension",
                type="number",
                default=1,
                description="Dimension along which Softmax will be computed"
            )
        ]

File: packages/core/src/lib/nodes/definitions/pytorch/softmax.ts
/**
 * PyTorch Softmax Activation Node Definition
 */

import { PassthroughNodeDefinition } from '../../base'
import { NodeMetadata, BackendFramework } from '../../contracts'
import { TensorShape, BlockConfig, ConfigField } from '../../../types'

export class SoftmaxNode extends PassthroughNodeDefinition {
  readonly metadata: NodeMetadata = {
    type: 'softmax',
    label: 'Softmax',
    category: 'activation',
    color: 'var(--color-destructive)',
    icon: 'Percent',
    description: 'Softmax activation',
    framework: BackendFramework.PyTorch
  }

  readonly configSchema: ConfigField[] = [
    {
      name: 'dim',
      label: 'Dimension',
      type: 'number',
      default: -1,
      description: 'Dimension along which to apply'
    }
  ]
}
